crates/runtime-impl/src/runtime.rs

```diff
@@ -70,26 +73,26 @@ fn compute_transition_attestation<P: MohoProgram>(
     );
 
     // Compute the transition.
-    let post_state = compute_transition::<P>(pre_state_ref, pre_moho_state, pre_inner_state, input);
+    transition_moho_state::<P>(pre_state_ref, moho_state, pre_inner_state, input);
 
     // Construct the attestation.
-    let post_state_commitment = compute_moho_state_commitment(&post_state);
+    let post_state_commitment = compute_moho_state_commitment(moho_state);
     StateRefAttestation::new(input_ref, post_state_commitment)
 }
 
 /// Computes and verifies a transition against an attestation we are trying to
 /// prove.
-pub fn compute_transition<P: MohoProgram>(
+pub fn transition_moho_state<P: MohoProgram>(
     pre_state_ref: &StateReference,
-    pre_moho_state: &MohoState,
+    moho_state: &mut MohoState,
     pre_inner_state: &P::State,
     input: &P::StepInput,
-) -> MohoState {
+) {
     // Check the pre-state matches that in the moho pre-state.
     let computed_inner_state_root = P::compute_state_commitment(pre_inner_state);
     assert_eq!(
         computed_inner_state_root,
-        pre_moho_state.inner_state(),
+        moho_state.inner_state(),
         "runtime: input moho state mismatch"
     );
```

Comment:
I see the value in doing this but it feels like maybe the extra safety check should be on the surrounding code so we don't have to do this hashing twice inside the proof.